src/main/java/com/yahoo/bullet/storm/JoinBolt.java

```diff
@@ -203,13 +174,13 @@ private void emitRetired(Map<String, AggregationQuery> forceEmit) {
         retireQueries().forEach(bufferedQueries::put);
     }
 
-    private boolean canEmit(String id, AggregationQuery query, Tuple returnTuple) {
+    private boolean canEmit(String id, AggregationQuery query, Tuple queryTuple) {
```

Comment:
I think this whole concept can be trimmed. There is no way we can get a query without the metadata for it, so queryTuple shouldn't be null at all (in fact, if it was null, we should throw away the query at creation time since we can't ever return any results for it and log an error).